src/mon/Paxos.cc

```diff
@@ -503,27 +503,25 @@ void Paxos::handle_last(MonOpRequestRef op)
   ceph_assert(g_conf()->paxos_kill_at != 2);
 
   // is everyone contiguous and up to date?
-  for (auto p = peer_last_committed.begin();
-       p != peer_last_committed.end();
-       ++p) {
-    if (p->second + 1 < first_committed && first_committed > 1) {
+  for (const auto& [peon_rank, peon_last_committed] : peer_last_committed) {
+    if (peon_last_committed + 1 < first_committed && first_committed > 1) {
       dout(5) << __func__
-	      << " peon " << p->first
-	      << " last_committed (" << p->second
+	      << " peon " << peer_rank
+	      << " last_committed (" << peon_last_committed
 	      << ") is too low for our first_committed (" << first_committed
 	      << ") -- bootstrap!" << dendl;
       op->mark_paxos_event("need to bootstrap");
       mon->bootstrap();
       return;
     }
-    if (p->second < last_committed) {
+    if (peon_last_committed < last_committed) {
       // share committed values
-      dout(10) << " sending commit to mon." << p->first << dendl;
+      dout(10) << " sending commit to mon." << peer_rank << dendl;
       MMonPaxos *commit = new MMonPaxos(mon->get_epoch(),
 					MMonPaxos::OP_COMMIT,
 					ceph_clock_now());
-      share_state(commit, peer_first_committed[p->first], p->second);
-      mon->send_mon_message(commit, p->first);
+      share_state(commit, peer_first_committed[peer_rank], peon_last_committed);
```

Comment:
and here.

Reply:
@tchaikov 
Sorry for these build errors. I haven't built them locally. 
Currently, I'm studying the Paxos protocol in Ceph to add one new PaxosService in Monitor.